common/session.go

```diff
@@ -32,9 +32,13 @@ import (
 	k6lib "go.k6.io/k6/lib"
 )
 
-// Ensure Session implements the EventEmitter and Executor interfaces
-var _ EventEmitter = &Session{}
-var _ cdp.Executor = &Session{}
+var _ session = &Session{}
+
+type session interface {
+	EventEmitter
+	cdp.Executor
+	ID() target.SessionID
+}
```

Comment:
@imiric Suggestion: Instead of making a Java-style preemptive interface, I think we better make the network manager accept only what it needs as I did in the browser:

https://github.com/grafana/xk6-browser/blob/112193c09aa304e6a33dbdee71c10e5776faaa82/common/browser.go#L61-L67

Reply:
Or, we can create an interface for each component like:

```golang
type ExecuteListener interface {
    EventEmitter
    cdp.Executor
}
```

As far as I see, this is the core behavior they all use.

Reply:
:man_shrugging: What's wrong with the current approach? Yes, it's mostly a useless interface that only has a single implementation, and it was only introduced so that it could be mocked in tests, but I don't see much harm with it.

I can't say that I'm a fan of defining anonymous types like that outside of tests. We might want to reuse `session` elsewhere where `*Session` is used now, and it would be easy to miss that it's defined inline only for `NetworkManager`.

> As far as I see, this is the core behavior they all use.

We also need `ID()`, though I didn't mean to change it everywhere else. We can get to making cleaner and reusable interfaces when we split `common`?

Reply:
> We can get to making cleaner and reusable interfaces when we split common?
I believe in starting refactoring here and there, little by little, instead of a significant refactoring which would be more challenging than today or in the future.

> What's wrong with the current approach?
What if another test wants to test something else? Are we going to add more methods to the interface and break all the other tests depending on it? This problem is the first one that comes to my mind right now.

Also, the Session concept has now two interfaces (_I don't mean Go interfaces :)_): `*Session` and `session,` which is confusing to me.

I believe code should change when necessary not for the future stuff that we can't see clearly from this moment. Right now, only the `NetworkManager` uses `session`, and another part of the code depends on `*Session`, increasing the complexity further.

But instead, if code had only accepted what it needs, all of these problems would evaporate, and we can keep refactoring the existing code without all these problems.

Reply:
> I believe code should change when necessary not for the future stuff that we can't see clearly from this moment.

Exactly why I'm against trying to design clean interfaces when we don't have a clear need for them yet. Doing this prematurely without specific requirements is guesswork at best, but it influences future changes once they become necessary.

> Right now, only the `NetworkManager` uses `session`, and another part of the code depends on `*Session`, increasing the complexity further.

Yeah, like I said, it was only added to be able to test `NetworkManager`. I don't want to get into a major refactor when it's not needed.

But I agree that the current `session` interface is sloppy. I'll make some changes based on your feedback and we can discuss it in a PR.

Reply:
The difference between `session` and `ExecuteEmitter` (or whatever or anonymous one) is that the current concrete implementations are let us discover the latter (`ExecuteEmitter`). I don't believe it's a premature abstraction. Alright, let's discuss afterward.